docs/index.asciidoc

```diff
@@ -154,9 +154,74 @@ The following is a basic {esql} query that sets the food name to transaction eve
 
 Set `config.support_escapes: true` in `logstash.yml` if you need to escape special chars in the query.
 
-In the result event, the plugin sets total result size in `[@metadata][total_values]` field. It also limits the result size to 1 when `FROM` query is used.
+In the result event, the plugin sets total result size in `[@metadata][total_values]` field.
+
+[id="plugins-{type}s-{plugin}-esql-event-mapping"]
+===== Mapping {esql} result to {ls} event
+{esql} returns query results in a structured tabular format, where data is organized into _columns_ (fields) and _values_ (entries).
+The plugin maps each value entry to an event, populating corresponding fields.
+For example, a query might produce a table like:
+
+[cols="2,1,1,1,2",options="header"]
+|===
+|`timestamp` |`user_id` | `action` | `status.code` | `status.desc`
+
+|2025-04-10T12:00:00 |123 |login |200 | Success
+|2025-04-10T12:05:00 |456 |purchase |403 | Forbidden (unauthorized user)
+|===
+
+For this case, the plugin creates two JSON look like objects as following:
+[source, json]
```

Comment:
This wording can be misleading, giving the user the idea they get "1 (Logstash) event per value entry", while it's not the case, where this filter with an ES|QL will always produce a single Logstash event.